### ISSUE TITLE
Add Series Accessor

### ISSUE THREAD

**Issue by tamargrey** [alteryx/woodwork#492]
- Creates the class `WoodworkColumnAccessor` 
- Gives access to fields such as `logical_type`, `semantic_tags`, `description`, etc on the Accessor
    - Note that this means that many `@property` fields should be defined on the Accessor so that users don't always have to go through the dictionary to access a column's typing information: `series.ww.logical_types` vs `series.ww.column_dict['logical_types']`

```python
@pd.api.extensions.register_series_accessor('ww')
class WoodworkColumnAccessor:
    def __init__(self, series): # series.ww
        self._series = series
        self._schema = None # Using "schema" to refer to column's typing Info

    def init(self, name=None, logical_type=None, **kwargs): # series.ww.init
        # validate params
        # confirm name matches series, or update to match passed in name
        # logic should be in DataColumn - throws ColumnNameMismatchWarning if it needs to be changed 
        
        logical_type = utils._get_logical_type(logical_type)
        
        # Need a way for this to happen in place or return a whole new series with accessor initialized
        _update_column_dtype(series, logical_type, name, inplace=True)
        
        self._schema = schema_column._get_column_dict(name, logical_type, **kwargs)
```

**Comment by tamargrey** [comment]
When initializing the Series Accessor, we want to update the underlying data to reflect the dtype changes (int46 -> Int64, string -> datetime) , or, in the case of LatLongs, the correct tuple formatting. For the Series Accessor, we'll want to do this inplace.

In the Table Accessor we use `_update_column_dtype`, but we don't care if the schema object changes since updating a column in a dataframe doesn't can be done without creating a new DataFrame object. But for the Series Accessor, if we can't do it in place, then `series` wouldn't match `series.ww._series`, and that will be a problem. 

We use `pd.Series.apply`, `pd.Series.astype`, and `pd.to_datetime(series)` which do not have inplace options, as far as I can tell. (not to mention the Dask and Koalas implementations which sometimes create an intermediate pandas object).

If we can't do this in place, we might need to come up with a solution that returns the new series that has the correct schema initialized, creating this kind of behavior:

```python
series = pd.Series(['1, 2', '3, 4'])
series2 = series.ww.init(logical_type=LatLong)

assert series.iloc[0] == '1, 2'
assert series.ww.schema is None

assert series2.iloc[0] == (1, 2)
assert series.ww.logical_type == LatLong
```

**Comment by thehomebrewnerd** [comment]
Closed by #562
